packages/discovery-provider/src/api/v1/users.py

```diff
@@ -3169,3 +3170,33 @@ def get(self, receiving_user_id, grantor_user_id):
                 return success_response(None)
 
             return success_response(email_access.to_dict())
+
+
+# Comments
+user_comments_response = make_response(
+    "user_comments_response", ns, fields.List(fields.Nested(base_comment_model))
+)
+
+
+@ns.route("/<string:id>/comments")
+class UserComments(Resource):
+    @record_metrics
+    @ns.doc(
+        id="""User Comments""",
+        description="""Get user comment history""",
+        params={"id": "A User ID"},
+        responses={
+            200: "Success",
+            400: "Bad request",
+            500: "Server error",
+        },
+    )
+    @ns.expect(pagination_with_current_user_parser)
+    @ns.marshal_with(user_comments_response)
+    @cache(ttl_sec=5)
+    def get(self, id):
+        args = pagination_with_current_user_parser.parse_args()
+        decoded_id = decode_with_abort(id, ns)
+        current_user_id = args.get("user_id")
+        user_comments = get_user_comments(args, decoded_id, current_user_id)
```

Comment:
nit: prefer we put the target user id and current user id into the args object itself.